rclcpp/src/rclcpp/service.cpp

```diff
@@ -39,11 +37,11 @@ ServiceBase::~ServiceBase()
 std::string
 ServiceBase::get_service_name()
 {
-  return this->service_name_;
+  return rcl_service_get_service_name(service_handle_);
```

Comment:
Alternatively, update the other locations where this is happening in this pr. Basically, I don't want an inconsistent state.